### ISSUE TITLE
Fonts do not load when specifying current project as package 

### ISSUE THREAD

**Issue by esDotDev** [flutter/flutter#44379]
Currently the way that Images and Fonts handle the package parameter as inconsistent. 

When using images, you can define the image like so: 
assets:
    - packages/demo1/images/logo.png

Then, the image will load fine with the: Image.asset("images/logo.png", package: "demo1"). 

However, fonts works differently. If I define a font as 
asset: packages/demo1/fonts/Poppins-Bold.ttf

And use: TextStyle(fontFamily: "packages/demo1/Poppins", package: "demo1")

The font does not render correctly in the root project ("demo1"). If I reference this package from another package, then the font **will** load correctly.

So, to recap:
* Images w/ package specifier work in their own package, and as dependancies. 
* Fonts w/ package specifier, do not work in their own package, but do work as dependancies

**Comment by cbracken** [comment]
@esDotDev thanks for filing this request. Here's our documentation on how to add additional fonts: https://flutter.dev/docs/cookbook/design/fonts

Do those instructions solve your issue? If not, could you provide some detail on the problems you're experiencing with the current behaviour?

**Comment by esDotDev** [comment]
Well in our use case, we had standalone apps, that were to be embedded in another app, and they need to work in both modes. We had to add special logic that injects that package name into fonts when the app was nested, but to omit them when not.

With images, we were able to just always specify package, and they worked when nested, and when standalone.

So, the request is to change Font behavior to match Images. If I specify the package for a font, that is the current package of the project, it should still work. Like images do currently.
